src/vm/virtualMachine.ts

```diff
@@ -83,9 +91,83 @@ export class VirtualMachine {
     if (op1Error !== undefined) {
       return { value: undefined, error: op1Error };
     }
-    const op1 = this.segments.memory.get(op1Addr);
+    let op1 = this.segments.memory.get(op1Addr);
 
-    return { value: undefined, error: undefined };
+    // If op0 is undefined, then we can deduce it from the instruction, dst and op1
+    // We also deduce the result based on the result logic
+    if (op0 === undefined) {
+      const { value: deducedValues, error: deducedOp0Error } = this.deduceOp0(
+        instruction,
+        dst,
+        op1
+      );
+      if (deducedOp0Error !== undefined) {
+        return { value: undefined, error: deducedOp0Error };
+      }
+      const [deducedOp0, deducedRes] = deducedValues;
+      if (deducedOp0 !== undefined) {
+        this.segments.memory.insert(op0Addr, deducedOp0);
+      }
+      res = deducedRes;
+    }
+
+    // If op1 is undefined, then we can deduce it from the instruction, dst and op0
```

Comment:
can we document in which cases op1 is undefined
also in comment I think we should use the full name of variables

Reply:
Honestly, I have no idea in which cases op1 can be undefined :/ I guess it could just be if the memory cell you read is empty